contrib/internal/httputil/trace.go

```diff
@@ -26,6 +26,7 @@ type TraceConfig struct {
 	Service        string                    // service name
 	Resource       string                    // resource name
 	QueryParams    bool                      // specifies that request query parameters should be appended to http.url tag
+	PathParams     map[string]string         // specifies framework-specific path parameters
```

Comment:
I'm not sure I understand the need for this. Isn't all of that already available in `req`?

Reply:
"High-level frameworks" such as Gin, Echo, Gorilla, Chi, etc. provide features to have endpoints with "path parameters" such as `/product/:id`, `/edit/:category/:id`.

So we introduced this new optional field so that "frameworks" based on `httputil.TraceAndServe()` can pass their path params.

Here is how we get them in [Gorilla](https://github.com/DataDog/dd-trace-go/pull/1106/files#diff-8106025518a24a34c7ff9cfc1048164b955d2fdfc3f8f8dc63deb78f1c45dfddR127), and in [Echo](https://github.com/DataDog/dd-trace-go/pull/1106/files#diff-d79ea332c1a2ff449d81f6083c88311a2eec2b1dcb64b6cde8e68f2535baac3fR25-R28).

Reply:
Ok, I see. Thanks for explaining.
```suggestion
	RouteParams     map[string]string         // specifies framework-specific route parameters (e.g. for route /user/:id coming in as /user/123 we'll have {"id": 123})
```
Do you mind if we go super explicit here? At first it didn't make sense what this is.

I suggest the word "route" instead of path because:
1. It matches gorilla/mux (maybe not echo so much)
2. One thinks of a route which usually has these type of parameters, and is not to be found in the terminology of the standard library (they use handler instead)
3. Path does not necessarily imply a smart routing system

What do you think?

Reply:
Good idea, thank you.